presto-main/src/main/java/com/facebook/presto/server/remotetask/SimpleHttpResponseHandler.java

```diff
@@ -79,6 +77,23 @@ else if (response.getStatusCode() == HttpStatus.SERVICE_UNAVAILABLE.code()) {
         }
     }
 
+    private String createErrorMessage(BaseResponse<T> response)
+    {
+        if (response instanceof JsonResponseWrapper) {
+            return format("Expected response code from %s to be %s, but was %s: %s%n%s",
+                    uri,
+                    OK.code(),
+                    response.getStatusCode(),
+                    response.getStatusMessage(),
+                    unwrapJsonResponse(response).getResponseBody());
+        }
+        return format("Expected response code from %s to be %s, but was %s: %s",
```

Comment:
Should we print the binary json (say as base64)?

Reply:
I am not sure how useful that would be for debugging.